e2e/next-sandbox/test/utils.ts

```diff
@@ -154,6 +154,25 @@ export async function waitForJson(
   );
 }
 
+export async function waitForTextContains(
+  oneOrMorePages: Page | Page[],
+  selector: IDSelector,
+  expectedValue: string,
+  options?: { timeout?: number }
+) {
+  const pages = Array.isArray(oneOrMorePages)
+    ? oneOrMorePages
+    : [oneOrMorePages];
+
+  return Promise.all(
+    pages.map((page) =>
+      expect(page.locator(selector)).toContainText(expectedValue, {
```

Comment:
`.toContainText` will do substring matching, whereas `.toHaveText` will do ful string matching. Typically I prefer the latter because it's more strict, but are you maybe using substring matching here deliberately for subdocs?